src/Fixer/ClassNotation/NoUnneededFinalMethodFixer.php

```diff
@@ -61,24 +67,33 @@ protected function applyFix(\SplFileInfo $file, Tokens $tokens)
 
             $classOpen = $tokens->getNextTokenOfKind($index, ['{']);
             $prevToken = $tokens[$tokens->getPrevMeaningfulToken($index)];
-            if (!$prevToken->isGivenKind(T_FINAL)) {
-                $index = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_CURLY_BRACE, $classOpen);
+            $classIsFinal = $prevToken->isGivenKind(T_FINAL);
 
-                continue;
-            }
-
-            $index = $this->fixClass($tokens, $classOpen);
+            $index = $this->fixClass($tokens, $classOpen, $classIsFinal);
         }
     }
 
     /**
      * @param Tokens $tokens
      * @param int    $classOpenIndex
+     * @param bool   $classIsFinal
      *
      * @return int
      */
-    private function fixClass(Tokens $tokens, $classOpenIndex)
+    private function fixClass(Tokens $tokens, $classOpenIndex, $classIsFinal)
     {
+        $methodVisibilities = [
+            T_PUBLIC,
+            T_PROTECTED,
+            T_PRIVATE,
+        ];
+
+        if (!$classIsFinal) {
+            $methodVisibilities = [
+                T_PRIVATE,
```

Comment:
`T_PRIVATE` is always part of the array, I would reverse the condition to initialize it with this value only, and add the others when the class is final.